lib/topological_inventory/orchestrator/metric_scaler.rb

```diff
@@ -1,46 +1,55 @@
 require "manageiq-loggers"
+require "topological_inventory/orchestrator/object_manager"
+require "topological_inventory/orchestrator/metric_scaler/watcher"
+require "topological_inventory/orchestrator/metric_scaler/prometheus_watcher"
 
 module TopologicalInventory
   module Orchestrator
     class MetricScaler
       attr_reader :logger
 
-      def initialize(logger = nil)
-        require "topological_inventory/orchestrator/metric_scaler/watcher"
+      def initialize(prometheus_hostname, logger = nil)
         @logger = logger || ManageIQ::Loggers::CloudWatch.new
+        @prometheus_hostname = prometheus_hostname
         @cache  = {}
       end
 
       def run
+        logger.info("#{self.class.name}##{__method__} Starting...")
         loop do
           run_once
 
           sleep 10
         end
+        logger.info("#{self.class.name}##{__method__} Complete")
       end
 
       def run_once
-        logger.info("#{self.class.name}##{__method__} Starting...")
         dc_names = object_manager.get_deployment_configs("metric_scaler_enabled=true").collect { |dc| dc.metadata.name }
 
         # newly_configured
-        (dc_names - @cache.keys).each { |name| @cache[name] = Watcher.new(name, logger).tap(&:start) }
+        (dc_names - @cache.keys).each { |name| @cache[name] = watcher_by_name(name).tap(&:start) }
         # no_longer_configured
         (@cache.keys - dc_names).each { |name| @cache.delete(name).stop }
         # currently configured
-        dc_names.each                 { |name| @cache[name].scale_to_desired_replicas }
-
-        logger.info("#{self.class.name}##{__method__} Complete")
+        dc_names.each                 { |name| @cache[name].scale_to_desired_replicas if @cache[name].scaling_allowed? }
       end
 
       private
 
-      def object_manager
-        @object_manager ||= begin
-          require "topological_inventory/orchestrator/object_manager"
```

Comment:
I'm thinking this was here for runtime require loading ( potential performance increase? ) I see this was moved to the top of the file.  What were the thoughts moving it to the top vs leaving it here?

Reply:
It was my intention, it makes file dependencies more clear when placed on the top of the file. 
Anyway this is loaded only once in both cases so I guess it won't have any impact to performance